### ISSUE TITLE
In .NET5+ projects one could use ProcessStartInfo.Arguments

### ISSUE THREAD

**Issue by drauch** [madelson/MedallionShell#95]
While I'm on it:

When targeting .NET5+ we could use ProcessStartInfo.ArgumentList to prevent argument injection by user input https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist?view=net-5.0

Of course this would need an #ifdef for older platforms not supporting this new property.

Best regards,
D.R.

**Comment by madelson** [comment]
@drauch yeah the need for #ifdef is why I've avoided this so far but I agree that it would be nice to leverage the native capability.

MedallionShell does encode arguments to prevent injection, though. Do you have reason to believe that the built-in argument encoding is not sufficient?

**Comment by drauch** [comment]
I haven't checked it to be honest.
